### PR TITLE
Building A Python Package and publishing it on PyPI

### PR DESCRIPTION
**Pull request by jamessandy**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tool:** [Grammarly](https://grammarly.com)
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by jamessandy** [comment]
This fixes #475

**Comment by jamessandy** [comment]
Done @lalith1403

**Comment by hectorkambow** [comment]
duplicate PR of #545

File: articles/building-a-python-package-and-publishing-on-pypi/index.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+      url=URL,
+      install_requires=INSTALL_REQUIRES,
+      packages=find_packages()
+      )
+
+```
+
+The first chunk of code imports all the tools we need to set up our library, the *VERSION* will be the release number for the package, the *PACKAGE NAME* will be a unique name you want to call your package, the *AUTHOR* and *AUTHOR EMAIL* will be your details and finally, the URL can be a site if you have one or maybe your GitHub Repo.
+
+If your package requires other packages to work then you’ll need to indicate them, for example, if you use other packages to write your code you should add them in the *INSTALL_REQUIRES* because your package depends on them (otherwise it will not work) but this package we just created doesn’t require any and I intentionally put wrote this note, in case you’re trying it out with your package idea.
```
Comment (lalith1403) [inline articles/building-a-python-package-and-publishing-on-pypi/index.md:120]: Rephrase the last sentence. Just mention that you have given an example for how to handle dependencies. No need to use the word intentionally.

File: articles/building-a-python-package-and-publishing-on-pypi/index.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+#### Packaging
+Now that we are done writing the code, we will set up all the needed files, and structure the codebase. Another important step will be to sign up for an account on [pypi.org](pypi.org), because that is where we will be hosting our package. After you sign up go to your terminal and pip install Twine with the command below.
```
Comment (lalith1403) [inline articles/building-a-python-package-and-publishing-on-pypi/index.md:123]: `pip install twine`. Wrap code using ``.

File: articles/building-a-python-package-and-publishing-on-pypi/index.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+```python
+pip install twine
+
+```
+
+The next thing you need to do is navigate to the directory of your project where the setup.py file is and then run this code.
+
+```python
+Python setup.py sdist bdist_wheel
```
Comment (lalith1403) [inline articles/building-a-python-package-and-publishing-on-pypi/index.md:133]: Is python in title case or lower case?

File: articles/building-a-python-package-and-publishing-on-pypi/index.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+```
+
```
Comment (lalith1403) [inline articles/building-a-python-package-and-publishing-on-pypi/index.md:135]: Include an image of the directory structure after the code is run.

File: articles/building-a-python-package-and-publishing-on-pypi/index.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+After running the code above you will notice more directories called *dist*, *build*, and *your_package.egg-info*. The *dist* is the most important right now because it holds the installation files that we will deploy to PyPI, open the directory and you’ll see two compressed files that are saved with *.tar* extension and also a wheel file.
```
Comment (lalith1403) [inline articles/building-a-python-package-and-publishing-on-pypi/index.md:136]: Explain what dist, build directories mean.

File: articles/building-a-python-package-and-publishing-on-pypi/index.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+To try out the test version you just created go to [test.pypi.org](https://test.pypi.org/) and check out the package, if you’re satisfied with what you see, then it is time to push it to the PyPi.
+
+```python
+Twine upload dist/*
+```
+
+Those are the steps required in building and deploying our package, you can run pip install <package name> to install it on your computer and try out the package using the code below.
+
+```python
+print(cal_lcm(4,5))
```
Comment (lalith1403) [inline articles/building-a-python-package-and-publishing-on-pypi/index.md:162]: Show the steps as follows;
1. Pip install calc_lcm
2. import calc_lcm
3. Then call the function.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
@jamessandy Great article. I found a few places where changes are required. Let me know once updated. Excited to see the updated article.

After making the changes, run the article through Grammarly once to avoid grammatical errors.

File: articles/building-a-python-package-and-publishing-on-pypi/index.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+VERSION = '0.0.0.1'
+PACKAGE_NAME = 'findlcm'
+AUTHOR = 'james'
+AUTHOR_EMAIL = 'james@sandy.com'
+URL = 'https://github.com/jamessandy/lcmfinder'
+
+LICENSE = 'MIT'
+DESCRIPTION = 'A package that helps find Lowest common multiple of a number'
+LONG_DESCRIPTION = (HERE / "README.md").read_text()
```
[the remark this answers is not in the document]
Comment (jamessandy) [reply articles/building-a-python-package-and-publishing-on-pypi/index.md:95]: it making refrence to the read me file created earlier

[the remark this answers is not in the document]
Comment (jamessandy) [reply articles/building-a-python-package-and-publishing-on-pypi/index.md:95]: *it is

File: articles/building-a-python-package-and-publishing-on-pypi/index.md (hunk 5 above)
Comment (jamessandy) [reply articles/building-a-python-package-and-publishing-on-pypi/index.md:136]: I've exchanged directory with folder, the I'veexplained dist,

**Comment by jamessandy** [comment]
All corrections made @lalith1403

**Review by lalith1403 — APPROVED** [review]

**Pull request closed without merging**